From ffc9c4649d5bb9569ada6f02ade75b68c54edb8b Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Fri, 1 Aug 2014 10:04:02 -0700
Subject: [PATCH 086/117] DO NOT UPSTREAM: Add missing v4l2 compat ioctls

SUBDEV_G_FMT and SUBDEV_G_FRAME_INTERVAL left out of the list and were
being ignored when issued by 32 bit clients.

Note that as of kernel 3.15, this code has been reworked so that the
compat handling defaults to the "native" ioctl.  This targetted patch
is for 3.14 only and should not be rebased on future kernels.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 drivers/media/v4l2-core/v4l2-compat-ioctl32.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/v4l2-core/v4l2-compat-ioctl32.c b/drivers/media/v4l2-core/v4l2-compat-ioctl32.c
index fca336b..a191f27 100644
--- a/drivers/media/v4l2-core/v4l2-compat-ioctl32.c
+++ b/drivers/media/v4l2-core/v4l2-compat-ioctl32.c
@@ -1091,6 +1091,8 @@ long v4l2_compat_ioctl32(struct file *file, unsigned int cmd, unsigned long arg)
 	case VIDIOC_ENUM_FREQ_BANDS:
 	case VIDIOC_SUBDEV_G_EDID32:
 	case VIDIOC_SUBDEV_S_EDID32:
+	case VIDIOC_SUBDEV_G_FMT:
+	case VIDIOC_SUBDEV_G_FRAME_INTERVAL:
 		ret = do_video_ioctl(file, cmd, arg);
 		break;
 
-- 
1.9.3

